app/questionnaire/navigation.py

```diff
@@ -50,8 +50,20 @@ def build_navigation(self, current_group_id, current_group_instance):
                 else:
                     logger.debug("building navigation", group_id=group['id'])
                     navigation.append(self._build_single_navigation(group, current_group_id, first_location))
+
+        if self._all_sections_completed(navigation):
```

Comment:
Just to be clear I don't mean we should add then hide things. I mean only add Summary or Confirmation type group/block if they have not got `"hide_in_navigation": true` *and* the questionnaire has be completed i.e. all blocks are visited